boards: stm32: Cleanup/Fix USB Kconfig option

galak opened the pull request:
The disco_l475_iot1 board should be using CONFIG_USB_DC_STM32 as the
Kconfig sybmol.

Also fix a minor typo in a comment in nucleo_f207zg with regards to
CONFIG_USB_DC_STM32

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
boards: stm32: Cleanup/Fix USB Kconfig option

The disco_l475_iot1 board should be using CONFIG_USB_DC_STM32 as the
Kconfig sybmol.

Also fix a minor typo in a comment in nucleo_f207zg with regards to
CONFIG_USB_DC_STM32

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

erwango commented on boards/arm/disco_l475_iot1/pinmux.c:
```diff
@@ -46,12 +46,12 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PA15, STM32L4X_PINMUX_FUNC_PA15_PWM2_CH1},
 #endif /* CONFIG_PWM_STM32_2 */
-#ifdef CONFIG_USB_STM32
+#ifdef CONFIG_USB_DC_STM32
```
Change seems fnie but what worrying me is that I actually tested initial code ok.
I should be able to test it again tomorrow, if this can wait. Even if there is no reason it would work as is.

erwango replied:
Reason it works "as is" is: at reset, default pinmux setting matches correct USB settings.
Looking into it in mode details

erwango replied:
after testing reference manual check, it appears that default gpio configuration and pin settings are not fully equivalent, but both works equally. Though, pushing an additional commit to remove PA9 setting which is not defined and not requested.

erwango committed:
boards: disco_l475_iot1: Remove PA9 setting for USB

PA9 was set as OTG_5V_VBUS. This definition is not available
and not required so removing it.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
